File: Python Course/Software Development Module - in Python/Week5/CinemaApp.py
"""
CinemaApp.py
Billy Cussen
27/09/2020
"""

#Objects
import gc
import Cinema
from Cinema import myCinema

#Variables
repeat = "y"

while repeat == "y":

    inputCheck = True

    while inputCheck:
        try:
            age = int(input("Please enter your Age: "))
        except:
            print ("You need to enter an Integer (e.g. 1, 10, 65)")
        else:
            if age < 0:
                print("You need to enter a Positive Number!")
            else:
                inputCheck = False

    myCinemaApp = myCinema(age)
    print(myCinemaApp.calculateTicketPrice())
    myCinemaApp = None
    gc.collect()

    repeat = ""
    while repeat != "y" and repeat != "n":
        repeat = str(input("Please enter if you would like to retry (\"Y\" or \"N\"): "))
        repeat = repeat.lower()
        if repeat != "y" and repeat != "n":
            print ("You need to enter \"Y\" or \"N\"")

print("Thanks for using my App!")

File: Java/Software Development Module - Java/Week5/CinemaApp.java
/*
CinemaApp.java
Billy Cussen
13/07/2020
*/

import java.util.Scanner;
import java.util.InputMismatchException;

public class CinemaApp {
    
    public static void main(String [] args){

        //Variables
        int age = 0;
        String result;
        boolean inputCheckCinema;

        //Object
        Scanner keyboard;
        Cinema myCinema;

        //Input
        do{

            inputCheckCinema = true;
            keyboard = new Scanner(System.in);

            try{

                System.out.print("Please enter your age: ");
                age = keyboard.nextInt();

                if(age < 0 || age > 117){
                    throw new InputMismatchException();
                }

            }catch(InputMismatchException e){
                System.out.println("You need to enter a number to denote your age!");
                inputCheckCinema = false;
            }

        }while(!inputCheckCinema);

        keyboard = null;

        //Process
        myCinema = new Cinema(age);
        myCinema.computePrice();

        //Output
        result = myCinema.getPrice();
        myCinema = null;
        System.out.println(result);
    }

}